### PR TITLE
MFLT-5099 Improve OTA-related documentation across all docs

### PR DESCRIPTION
**Pull request by fnune**
- Add new OTA-related terms to terminology and update existing
- Fix styling for auth token callout
- Update CLI docs with latest --help and update auth requirements table
- Update "terminology overview" section in all release guides
  - Mostly remove it, and link to relevant platform-agnostic docs
    - Remove redundant identifying software/binaries sections on OTA guides
- Realign platform-specific OTA titles
  - They're now in the format `$PLATFORM OTA Integration Guide`
- Link to SV/HV and terminology docs from platform OTA guide
- Update the Software Versions & Hardware Versions doc with OTA info

### CODE REVIEW

File: docs/android/android-releases-guide.mdx [hunk 1]
```diff
@@ -259,9 +212,9 @@ INFO: You can view in the UI here:
 
 :::
 
-### Deploy Release
+### Activating the Release
```
Comment (HBehrens) [inline docs/android/android-releases-guide.mdx:215]: should this be "a" Release? Same below?

File: docs/linux/linux-releases-guide.mdx [hunk 2]
```diff
@@ -15,75 +15,28 @@ with Memfault!
 
 ## Terminology Overview
 
-Before we get started, let's quickly recap some of the terminology used
-throughout the Memfault documentation and how it relates to release management.
+For a platform-agnostic introduction to Memfault related to OTA, check out
```
Comment (HBehrens) [inline docs/linux/linux-releases-guide.mdx:18]: this sounds as if you could choose to between a) the platform-agnostic version or b) continue reading for the platform-specific version. How about

> Please refer to [Over-the-Air Updates(OTA)](...) for an introduction to Memfault's OTA concepts and terminology. The paragraphs below discuss the platform-specific aspects for $platform.

File: docs/mcu/releases-guide.mdx [hunk 3]
```diff
@@ -1,98 +1,36 @@
 ---
 id: releases-integration-guide
-title: Releases Integration Guide
-sidebar_label: Release Management
+title: MCU OTA Integration Guide
+sidebar_label: OTA Management
```
Comment (HBehrens) [inline docs/mcu/releases-guide.mdx:4]: I ❤️ the change to "OTA"

File: docs/platform/ota.mdx [hunk 4]
```diff
@@ -37,6 +37,15 @@ platform:
 - [OTA Updates for Android (AOSP)][platform-android],
 - [OTA Updates for Embedded Linux][platform-linux].
 
+Read about basic Memfault-specific terminology that you may need to refer to to
```
Comment (HBehrens) [inline docs/platform/ota.mdx:40]: We should generally try to avoid terms like "basic", "simple", or "easy" unless they have actual behavior meaning (e.g. "basic filter rule vs. complex filter rule").

**Review by HBehrens — COMMENTED** [review]
Did a quick pass on the structure and added some inline comments – did not read the actual text.

File: docs/platform/ota.mdx (hunk 4 above)
Comment (fnune) [reply docs/platform/ota.mdx:40]: How about `prerequisite Memfault terminology`?

File: docs/android/android-releases-guide.mdx (hunk 1 above)
Comment (fnune) [reply docs/android/android-releases-guide.mdx:215]: No, this section is a step-by-step guide that guides the user through the activation of a specific fictitious Release.

File: docs/platform/ota.mdx (hunk 4 above)
Comment (fnune) [reply docs/platform/ota.mdx:40]: I went with `Read about related Memfault-specific terminology...`

File: docs/android/android-releases-guide.mdx [hunk 5]
```diff
@@ -271,9 +225,10 @@ specific cohort check out our [api-docs](https://api-docs.memfault.com/).
 
 :::
 
-In the UI navigate to "Fleet" -> "Cohorts". Click on the picker under next to
-the `default` cohort under "Target Release", select the Release to deploy and
-you will be prompted with options that can be performed for the release.
+In the application, navigate to **Fleet** -> **Cohorts**. Click on the picker
+under next to the `default` cohort under "Target Release", select the Release to
+activate and you will be prompted with options that can be performed for the
+Release.
```
Comment (HBehrens) [inline docs/android/android-releases-guide.mdx:231]: Is there a place we can link to in our general OTA docs for more details / is there an opportunity to shorting this section?

File: docs/ci/install-memfault-cli.mdx [hunk 6]
```diff
@@ -56,6 +56,7 @@ Commands:
   upload-mcu-symbols          Upload symbols for an MCU build.
   upload-ota-payload          Upload a binary to be used for an OTA update.
   upload-symbols              [DEPRECATED] Upload symbols for an MCU or...
+  upload-xed                  Upload an .xed or .xcd file for analysis.
```
Comment (HBehrens) [inline docs/ci/install-memfault-cli.mdx:59]: nit for future updates: I believe the updates for xed/xcd could have been a separate PR. Also, should we invest in automation to detect outdated docs for our CLI?

File: docs/linux/linux-releases-guide.mdx [hunk 7]
```diff
@@ -15,75 +15,29 @@ with Memfault!
 
 ## Terminology Overview
 
-Before we get started, let's quickly recap some of the terminology used
-throughout the Memfault documentation and how it relates to release management.
+Please refer to [Over-the-Air Updates(OTA)](/docs/platform/ota) for an
+introduction to Memfault's OTA concepts and terminology. The paragraphs below
+discuss the platform-specific aspects for Embedded Linux.
```
Comment (HBehrens) [inline docs/linux/linux-releases-guide.mdx:20]: Does it make sense to extract this as a component or not worth it?

File: docs/linux/linux-releases-guide.mdx [hunk 8]
```diff
@@ -15,75 +15,29 @@ with Memfault!
...
-  impacting that unit.
-- `Cohort` - A grouping of devices. You will be able to filter data and deploy
-  releases by this granularity. For example, you could use a cohort to represent
-  devices used for different release stages ("alpha", "beta", "prod") or to
-  identify devices by location ("office", "site A", "site B") or a combination
-  of both ("Site A Alpha", "Site A Beta", etc). A device is always assigned to
-  exactly one "Cohort" at any given time but can freely be moved to another at
-  any time.
-- `Fleet` - All devices within a Project
+- Device ID/Device Serial: a unique identifier of a device. For example, this
```
Comment (HBehrens) [inline docs/linux/linux-releases-guide.mdx:27]: Same as above: can we further de-emphasize device serial here?

File: docs/linux/linux-releases-guide.mdx [hunk 9]
```diff
@@ -15,75 +15,29 @@ with Memfault!
...
+  example, this can be used to represent the different stages of
+  [manufacturing builds](https://mflt.io/hardware-engineer-speak) (`evt`, `dvt`,
+  `pvt`, ...).
 
-### Identifying a Device
+:::note
 
-For a "Device" in a project, several pieces of identifying information are
-tracked.
+For any given device, neither of these pieces of information should ever change.
```
Comment (HBehrens) [inline docs/linux/linux-releases-guide.mdx:38]: Same as on Android, there are good reasons for them to change but we will need to explain the consequences.

File: docs/linux/linux-releases-guide.mdx [hunk 10]
```diff
@@ -218,14 +172,15 @@ specific cohort check out our [api-docs](https://api-docs.memfault.com/).
 
 :::
 
-In the UI navigate to "Fleet" -> "Cohorts". Click on the picker under next to
-the `default` cohort under "Target Release", select the Release to deploy and
-you will be prompted with options that can be performed for the release.
+In the application, navigate to **Fleet** -> **Cohorts**. Click on the picker
+under next to the `default` cohort under "Target Release", select the Release to
+activate and you will be prompted with options that can be performed for the
+Release.
```
Comment (HBehrens) [inline docs/linux/linux-releases-guide.mdx:178]: Same as for Android: Is there an opportunity to shorten this further by linking to the platform-agnostic docs?

File: docs/mcu/releases-guide.mdx [hunk 11]
```diff
@@ -1,98 +1,37 @@
...
-  impacting that unit.
-- `Cohort` - A grouping of devices. You will be able to filter data and deploy
-  releases by this granularity. For example, you could use a cohort to represent
-  devices used for different release stages ("alpha", "beta", "prod") or to
-  identify devices by location ("office", "site A", "site B") or a combination
-  of both ("Site A Alpha", "Site A Beta", etc). A device is always assigned to
-  exactly one "Cohort" at any given time but can freely be moved to another at
-  any time.
-- `Fleet` - All devices within a Project
+- Device Serial: a unique identifier of a device. For example, this may be a
```
Comment (HBehrens) [inline docs/mcu/releases-guide.mdx:21]: should this be "Device ID"?

File: docs/platform/software-version-hardware-version.mdx [hunk 12]
```diff
@@ -64,32 +63,26 @@ A Software Version is a single build of a particular Software Type. Software
...
   firmware, the firmware's Software Version is used to determine if a device
-  requires an update
+  requires an update.
 
 ### Version Schemes
 
-#### Ordering
+If you're interested in learning how Memfault decides which OTA Release to
+deliver to a specific device, refer to the
+[Updating Behavior section on our platform-agnostic OTA documentation page](/docs/platform/ota#updating-behavior).
```
Comment (HBehrens) [inline docs/platform/software-version-hardware-version.mdx:77]: I'd rephrase this to justify the interest in OTA at this point, e.g.

```suggestion
Comparison between versions is a fundamental input to Memfault's OTA system.
Consult the section
[Updating Behavior on our OTA documentation](/docs/platform/ota#updating-behavior) to see how the rules below are being applied.
```

**Review by HBehrens — APPROVED** [review]
Looks great! A few inline comments.

File: docs/linux/linux-releases-guide.mdx (hunk 7 above)
Comment (fnune) [reply docs/linux/linux-releases-guide.mdx:20]: IMO not worth it.

File: docs/linux/linux-releases-guide.mdx (hunk 10 above)
Comment (fnune) [reply docs/linux/linux-releases-guide.mdx:178]: Agree, but will postpone and change once we've merged the Version Matrix docs.

File: docs/android/android-releases-guide.mdx [hunk 13]
```diff
@@ -16,80 +16,34 @@ Memfault!
...
-  always represented by the `android-build` Software Type.
-- `Software Version` / `software_version` - Each build of a "software_type" is
-  uniquely identified by its "software_version".
-
-For more information, see our
-[Software Version](https://mflt.io/android-software-versions) documentation.
+:::note
 
-### Identifying Firmware Binaries
+For any given device, neither of these pieces of information should ever change.
```
[the remark this answers is not in the document]
Comment (fnune) [reply docs/android/android-releases-guide.mdx:44]: I'm finding it hard to write anything useful instead of this, that doesn't get too much into the weeds. It also depends on whether `For any given device` refers to a Memfault device or an actual device in the field. If it's the former, then the sentence stands.
